iocore/net/SSLUtils.cc

```diff
@@ -100,10 +102,97 @@ static int ssl_vc_index = -1;
 static ink_mutex *mutex_buf      = nullptr;
 static bool open_ssl_initialized = false;
 
-/* Using pthread thread ID and mutex functions directly, instead of
- * ATS this_ethread / ProxyMutex, so that other linked libraries
- * may use pthreads and openssl without confusing us here. (TS-2271).
- */
+TLSKeyLogger::TLSKeyLogger()
+{
```

Comment:
These simple ctor's etc., why not put them in the include file where they become implicitly inlined? I'd say anything short than a few simple lines should go there :-).

Reply:
My coding practice is pretty purist when it comes to implementation being in .cc files.  It's just a preference thing for me. Also, if it's in the header, changes often impact multiple translation units which increases build times.

But that's clearly just a personal preference of mine. If you ask again I'll put it in the header. Note that since this is a singleton, this constructor is only called once.

Thanks for reviewing and sharing your thoughts.

Reply:
I talked with Leif offline. He correctly points out that it's more consistent with the rest of the code base to have these smaller function definitions in the header. In keeping with this pattern, I moved these over to the header. 👍